src/inmanta/deploy/scheduler.py

```diff
@@ -675,37 +703,83 @@ async def report_resource_state(
                 self._deploying_latest.remove(resource)
+
+                previous_deployment_result = state.deployment_result
+
                 state.deployment_result = deployment_result
                 self._work.finished_deploy(resource)
+
+                # Set of dependant resources that are interested in deploys of the current resource
+                concerned_resources: set[ResourceIdStr] = set()
+                provides: Set[ResourceIdStr] = self._state.requires.provides_view().get(resource, set())
+
                 if deployment_result is DeploymentResult.DEPLOYED:
                     self._state.dirty.discard(resource)
+                    is_dirty = False
+
+                    if previous_deployment_result != DeploymentResult.DEPLOYED:
+                        # This resource went from not deployed to deployed
+                        # we have to inform its dependants regardless of event propagation
+                        # so that we can try to re-deploy them in case this resource's previous
+                        # failure was the reason they were skipped
+
+                        # FIXME: stop informing resource skipped because of
+                        #  a custom handler skip via this mechanism
+
+                        for dependant in provides:
+                            if not self._state.resources.get(dependant, None):
+                                continue
+                            if not (dependant_status := self._state.resource_state.get(dependant, None)):
+                                continue
+                            if dependant_status.blocked is BlockedStatus.NO:
+                                concerned_resources.add(dependant)
+
                 else:
                     # In most cases it will already be marked as dirty but in rare cases the deploy that just finished might
                     # have been triggered by an event, on a previously successful deployed resource. Either way, a failure
                     # (or skip) causes it to become dirty now.
                     self._state.dirty.add(resource)
+                    is_dirty = True
+
+                    if deployment_result is DeploymentResult.SKIPPED:
+                        # Also add back dependents to the dirty set
+                        # For now we add all dependents
+
+                        dependant_resources = {
+                            dependant for dependant in provides if self._state.resources.get(dependant, None) is not None
+                        }
+                        self._state.dirty.update(dependant_resources)
+                        for dependant in dependant_resources:
+                            # No point in re-trying dependants since this resource was skipped
+                            # FIXME as part of https://github.com/inmanta/inmanta-core/issues/8423
+                            self._timer_manager.remove_resource(dependant)
```

Comment:
I may be missing something (or have forgotten about a previous discussion) but I don't think this is correct / desirable. Could you give some more context as to why we'd want to do this?

Reply:
I think you're right. Here's the train of thought that lead to this:
I was only considering `repair` semantics and so since we know that the providing resource was just skipped, all dependents would fail their repair so we can cancel them right away. 
But when considering `deploy` semantic as well I don't think this is correct since this would destroy timers for deploy that could very well succeed.

Reply:
There's even more to it: while most dependents will skip as a result of this one skipping, that behavior is not guaranteed, so we have to give their handlers the opportunity to handle it how they see fit, i.e. we need to deploy them as planned.

To be clear, I was talking about this whole block, not just the timer part.

Reply:
I couldn't find a good reason for why this was added, removed.